### DISCUSSION TITLE
Xonsh crashes after vox fails

### DISCUSSION THREAD

**Discussion by JanczarKurek** [xonsh/xonsh#5731] [Q&A] [answered]
Hi! Im not sure whether it's an issue with xonsh or with vox, but I installed vox recently using https://aur.archlinux.org/packages/xontrib-vox-git and when I run any vox command that results with an error, e.g. `vox list` with no environments or `vox new` then vox nicely prints an error, e.g. `vox new: error: the following arguments are required: ENV` and then whole xonsh crashes with exit status `2`. Any ideas on how to avoid this situation?

**Comment by JanczarKurek** [comment]
Ok I get it now - one can just sys.exit from a xonsh plugin and it exits whole xonsh and vox just uses Argparser in the version I installed, funny.

**Reply by anki-code** [reply, marked as answer]
It should be fixed in [0.18.4](https://github.com/xonsh/xonsh/releases/tag/0.18.4): Now SystemExit that invoked from callable alias follows to exiting from callable alias instead of exiting the entire shell.
